src/pages/organize/[orgId]/campaigns/index.tsx

```diff
@@ -19,7 +41,31 @@ export const getServerSideProps : GetServerSideProps = scaffold(async (ctx) => {
     await ctx.queryClient.prefetchQuery(['org', orgId], getOrg(orgId as string, ctx.apiFetch));
     const orgState = ctx.queryClient.getQueryState(['org', orgId]);
 
-    if (orgState?.status === 'success') {
+    await ctx.queryClient.prefetchQuery(['campaigns', orgId], getCampaigns(orgId as string, ctx.apiFetch));
+    const campaignsState = ctx.queryClient.getQueryState(['campaigns', orgId]);
+
+    await ctx.queryClient.prefetchQuery(['upcomingEvents', orgId], getUpcomingEvents(orgId as string, ctx.apiFetch));
+    const upcomingEventsState = ctx.queryClient.getQueryState(['upcomingEvents', orgId]);
+
+    await ctx.queryClient.prefetchQuery(['events', orgId], getEvents(orgId as string, ctx.apiFetch));
+    const eventsState = ctx.queryClient.getQueryState(['events', orgId]);
+
+    await ctx.queryClient.prefetchQuery(['surveys', orgId], getSurveys(orgId as string, ctx.apiFetch));
+    const surveysState = ctx.queryClient.getQueryState(['surveys', orgId]);
+
+    await ctx.queryClient.prefetchQuery(['calls', orgId], getAllCallAssignments(orgId as string, ctx.apiFetch));
+    const allCallAssignmentsState = ctx.queryClient.getQueryState(['calls', orgId]);
+
+    await ctx.queryClient.prefetchQuery(['canvasses', orgId], getCanvasses(orgId as string, ctx.apiFetch));
+    const canvassesState = ctx.queryClient.getQueryState(['canvasses', orgId]);
+
+    await ctx.queryClient.prefetchQuery(['activities', orgId], getActivities(orgId as string, ctx.apiFetch));
+    const activitiesState = ctx.queryClient.getQueryState(['activities', orgId]);
+
+    await ctx.queryClient.prefetchQuery(['locations', orgId], getLocations(orgId as string, ctx.apiFetch));
+    const locationsState = ctx.queryClient.getQueryState(['locations', orgId]);
+
+    if (orgState?.status === 'success' && campaignsState?.status === 'success' && eventsState?.status === 'success'  && allCallAssignmentsState?.status === 'success' && canvassesState?.status === 'success' && surveysState?.status === 'success' && upcomingEventsState?.status === 'success' && activitiesState?.status === 'success' && locationsState?.status === 'success') {
```

Comment:
Honestly, I'm wondering whether we should even pre-load this at all, or only load it on the client side and show a loading indicator instead. What do you think?

Reply:
I can't see any advantage to client-side except for less ridiculous looking code? 
We don't need SEO or anything here but wouldn't prefetching be a nicer UX?
Is there something else to consider?

Reply:
I think what's the nicer UX depends on the goal of the user. I totally get your point for many use cases, but consider the following:

A user wants to edit an existing campaign. They navigate to the campaigns dashboard, and click the campaign they want to edit.

*UX when pre-loading everything (current)*: When navigating to the campaigns dashboard, the server will pre-load all the data, which takes a few seconds.

*UX when pre-loading campaigns only*: When navigating to the campaigns dashboard, only the campaigns are preloaded, so the time to first paint is faster.

There is a downside to going full client-side though, and it's that making the requests from the client is slower, and that's multiplied by the fact that it needs to make multiple requests (for surveys, call assignments, events etc). The solution to that can be to create a single API endpoint in NEXT which makes all the necessary requests to the Zetkin API and returns the data in bulk.

But I guess all of this doesn't really make much sense since we're hiding this temporarily anyway. 😊

Reply:
aha, yes, i understand. Maybe better to make another issue to do this across multiple pages?

Reply:
Yes, I'll do that! :)

Reply:
Logged as #238